src/components/QuestionComponent/index.js

```diff
@@ -1,11 +1,29 @@
+import {nanoid} from 'nanoid';
 import {useState} from 'react';
 
+import {useStore} from '../../hooks/useStore';
 import Form from '../Form';
 
 import StyledQuestionComponent from './styled';
 
-export default function QuestionComponent({question, addAnswer}) {
+export default function QuestionComponent({question}) {
 	const [toggle, setToggle] = useState(true);
+	const questions = useStore(state => state.questions);
+	const addNewAnswer = useStore(state => state.addNewAnswer);
+
+	const addAnswer = (event, id) => {
+		event.preventDefault();
+		const index = questions.findIndex(question => {
```

Comment:
Better solution would be possible with map instead your solution from line 16 to 24

Reply:
Done :)